timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/MultiNodePaxosTimeLockServerIntegrationTest.java

```diff
@@ -64,6 +78,50 @@ public void bringAllNodesOnline() {
     }
 
     @Test
+    public void lockRequestThrows500DueToLeaderElection() throws InterruptedException {
+        CLUSTER.lock(CLIENT_1, BLOCKING_LOCK_REQUEST);
+
+        TestableTimelockServer leader = CLUSTER.currentLeader();
+        Executors.newSingleThreadExecutor().submit(() -> {
+            assertThatThrownBy(() -> leader.lock(CLIENT_2, BLOCKING_LOCK_REQUEST))
+                    .isInstanceOf(InterruptedException.class);
```

Comment:
does this actually result in an `InterruptedException` locally?
also note that if this assertion fails, it won't fail the test, because it's running on another thread